### PR TITLE
[DOXIA-695] Support heading level 6

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

File: doxia-sink-api/src/main/java/org/apache/maven/doxia/sink/Sink.java [hunk 1]
```diff
@@ -503,7 +510,7 @@ public interface Sink extends AutoCloseable {
 
     /**
      * Starts a 5th heading element which contains the topic of the section.
-     * This has to be contained within a {@link #section4()} element.
+     * This has to be contained within a {@link #section5()} element.
```
Comment (michael-o) [inline doxia-sink-api/src/main/java/org/apache/maven/doxia/sink/Sink.java:513]: It was a typo, wasn't it?

**Comment by michael-o** [comment]
What about formats which do not support level 6, but it requested from code? E.g. Apt?

File: doxia-modules/doxia-module-apt/src/test/java/org/apache/maven/doxia/module/apt/AptSinkTest.java [hunk 2]
```diff
@@ -126,6 +126,11 @@ protected String getSection5Block(String title) {
         return EOL + StringUtils.repeat(AptMarkup.SECTION_TITLE_START_MARKUP, 4) + title + EOL + EOL + EOL;
     }
 
+    /** {@inheritDoc} */
+    protected String getSection6Block(String title) {
+        return EOL + StringUtils.repeat(AptMarkup.SECTION_TITLE_START_MARKUP, 4) + title + EOL + EOL + EOL;
```
Comment (michael-o) [inline doxia-modules/doxia-module-apt/src/test/java/org/apache/maven/doxia/module/apt/AptSinkTest.java:131]: 4, not 5?

File: doxia-modules/doxia-module-xhtml5/src/main/java/org/apache/maven/doxia/module/xhtml5/Xhtml5Parser.java [hunk 3]
```diff
@@ -158,7 +158,7 @@ protected void handleEndTag(XmlPullParser parser, Sink sink)
         } else if (parser.getName().equals(TITLE.toString())) {
             sink.title_();
         } else if (parser.getName().equals(BODY.toString())) {
-            consecutiveSections(0, sink, null);
+            emitHeadingSections(0, sink, false);
```
Comment (michael-o) [inline doxia-modules/doxia-module-xhtml5/src/main/java/org/apache/maven/doxia/module/xhtml5/Xhtml5Parser.java:161]: This one confuses me

**Review by michael-o — CHANGES_REQUESTED** [review]

File: doxia-modules/doxia-module-apt/src/test/java/org/apache/maven/doxia/module/apt/AptSinkTest.java (hunk 2 above)
Comment (kwin) [reply doxia-modules/doxia-module-apt/src/test/java/org/apache/maven/doxia/module/apt/AptSinkTest.java:131]: Yes, APT only supports up to level 5 and converts everything above to level  5

File: doxia-modules/doxia-module-xhtml5/src/main/java/org/apache/maven/doxia/module/xhtml5/Xhtml5Parser.java (hunk 3 above)
Comment (kwin) [reply doxia-modules/doxia-module-xhtml5/src/main/java/org/apache/maven/doxia/module/xhtml5/Xhtml5Parser.java:161]: This is just about replacing the deprecated method with the non deprecated one.

File: doxia-modules/doxia-module-apt/src/test/java/org/apache/maven/doxia/module/apt/AptSinkTest.java (hunk 2 above)
Comment (michael-o) [reply doxia-modules/doxia-module-apt/src/test/java/org/apache/maven/doxia/module/apt/AptSinkTest.java:131]: This needs a comment

File: doxia-modules/doxia-module-xhtml5/src/main/java/org/apache/maven/doxia/module/xhtml5/Xhtml5Parser.java (hunk 3 above)
Comment (michael-o) [reply doxia-modules/doxia-module-xhtml5/src/main/java/org/apache/maven/doxia/module/xhtml5/Xhtml5Parser.java:161]: Alright.

File: doxia-modules/doxia-module-apt/src/test/java/org/apache/maven/doxia/module/apt/AptSinkTest.java (hunk 2 above)
Comment (kwin) [reply doxia-modules/doxia-module-apt/src/test/java/org/apache/maven/doxia/module/apt/AptSinkTest.java:132]: Done

**Review by michael-o — APPROVED** [review]

**Pull request merged**